recipes/lbfgspp/meta.yaml

```diff
@@ -0,0 +1,42 @@
+{% set name = "lbfgspp" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/yixuan/lbfgspp/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 7101744a538c3aff52e10c82267305847b0b5e9d39f9974b4b29812cd1398ff9
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - eigen
+
+test:
+  commands:
+    - test -f $PREFIX/include/LBFGS.h  # [unix]
+    - test -f $PREFIX/include/LBFGSB.h  # [unix]
+    - test -d $PREFIX/include/LBFGSpp  # [unix]
+    - if not exist %LIBRARY_PREFIX%\include\LBFGS.h exit 1  # [win]
+    - if not exist %LIBRARY_PREFIX%\include\LBFGSB.h exit 1  # [win]
```

Comment:
Could you please also compile and run a test program in the test section from the examples to check that this library is compatible with our compilers? Sometimes our compilers are too old and require source patches to silence syntax errors, etc.